api/src/controllers/profile.ts

```diff
@@ -229,38 +255,38 @@ export const updateProfileQuotaSize = async (
 ): Promise<void> => {
   const { ProfileModel } = dm;
   const { profileId } = params;
-  const { requestedQuotaSize } = body;
+  const { requestedQuotaSize, namespace } = body;
 
   try {
     const profile: ProjectProfile = await ProfileModel.findById(profileId);
-    const quotaSize: ProjectQuotaSize = await getQuotaSize(profile);
+    const namespaceQuotaSize = await getNamespaceQuotaSize(profile, namespace);
     const allowedQuotaSizes: NameSpacesQuotaSize =
-      getAllowedQuotaSizes(quotaSize);
+      getAllowedQuotaSizes(namespaceQuotaSize);
     const requiresHumanAction = true;
 
     // verify if requested quota size is valid
     Object.keys(requestedQuotaSize).forEach((key) => {
       if (
         !(
           requestedQuotaSize[key] &&
-          (quotaSize[key].includes(requestedQuotaSize[key]) ||
+          (namespaceQuotaSize[key].includes(requestedQuotaSize[key]) ||
             allowedQuotaSizes[key].includes(requestedQuotaSize[key]))
         )
       ) {
         throw new Error("Please provide correct requested quota size in body");
       }
     });
```

Comment:
imo this could be abstracted away into a function so that it is more sensical to read. No change needed